### ISSUE TITLE
REFPROP applying density limits to Peng-Robinson calculations

### ISSUE THREAD

**Issue by nist-aharvey** [usnistgov/REFPROP-issues#347]
I chose ethylene glycol as the fluid, and told it to use Peng-Robinson for all calculations (problem happens with or without the volume translation).
I get error messages for the density being out of range, so for example it won't calculate V/L saturation at 323 K.
I believe the purpose of these density limits is to keep from getting into places where the complicated Helmholtz EOS was not fitted and might behave badly.  But if you are using a cubic like Peng-Robinson instead of the default EOS this doesn't matter, and there is no reason not to go ahead and do the calculations just because P-R gives densities that are higher than reality.
I think this maximum density check should be disabled if possible when the Peng-Robinson option is chosen.

**Comment by KoldMen** [comment]
Some density values can lead to negative pressures. Some calculations can fail when p < 0, for example in the phiderv routine. Removing the limits can result in some breaking changes. May be best to postpone it to next major revision.

**Comment by nist-aharvey** [comment]
I'm sure the only way to get p<0 from the Peng-Robinson equation when exceeding the MAXIMUM density is if you get the density so high that the molar volume is .LE. the "b" parameter.  So maybe the b parameter should provide a new maximum density?
Actually, using the maximum density from the fundamental EOS could get you in trouble in the other direction, if you had a fluid where P-R *under*estimated the liquid density.  In principle you could end up with your liquid molar volume hitting the b parameter and the PR EOS blowing up even though you are within the density limits of the fundamental EOS. 

Still might be best to postpone a fix, especially if nobody else has complained about it, but I don't think the simple p<0 argument is quite right.

**Comment by ianhbell** [comment]
I agree with @nist-aharvey, the density limit when PR is enabled should be that associated with the covolume of the cubic EOS.

**Comment by KoldMen** [comment]
@ianhbell Agree with you and @nist-aharvey. Limits routine should be changed for P-R. 

@nist-aharvey I was only saying that the fix should be postponed to next major revision when other breaking changes are also addressed. In any case when @ianhbell's recent Isochoric routines are added, fail on p < 0 may have to be changed to warn on p < 0.  What do you say @ianhbell ?

Coming back to the original question, should one even try using P-R to estimate properties of ethylene glycol ??

**Comment by ianhbell** [comment]
@KoldMen I would say it depends on the use case.  If you need "a number", then maybe, but if you are trying to get quantitative agreement, then no.

**Comment by nist-aharvey** [comment]
I agree that P-R would not be a good choice for ethylene glycol properties.  I don't remember now how I stumbled upon the bug; I sort of think maybe I was reviewing a paper that did P-R calculations on mixtures with EG.  Or possibly I was looking at the recent EG viscosity paper and wanted to do some EOS calculations for some reason.

But I think P-R often gives liquid densities that are off (especially without volume translation), so I could envision it happening for other fluids (I just tried 6 or 8 others and got it to happen for chlorine).  And if the liquid density is off in the other direction, you could potentially get the related problem where an input density was allowed that was out of the physically reasonable range for P-R (determined by the covolume b).  
But since P-R isn't really a core functionality of REFPROP, this bug is probably not very high on the priority list.

**Comment by ianhbell** [comment]
I think we need to prioritize other things so this is going to go into the wontfix category

**Comment by nist-aharvey** [comment]
I'm fine with this, assuming "wontfix" means for the next version (as opposed to never). In the long run, if REFPROP is going to offer Peng-Robinson calculations as an option it shouldn't fail at conditions where there is no good reason for failure.
